### ISSUE TITLE
Sluggish

### ISSUE THREAD

**Issue by YourIssue** [JapaMala/armok-vision#55]
Is Armok Vision suppose to be very sluggish in following dwarf fortress? FPS wise Armok Vision is fast but following the DF screen there is a decent amount delay. If this isn't how it should be then I guess this post belongs here for now and if otherwise then I'll remove it, thank you.

**Comment by JapaMala** [comment]
Yeah, the latest version decreases the update speed from 30 times per second down to once per second. This means that it only checks where you're looking in DF every second, rather than every frame.

I'll look into separating things that don't need fast updates from things that need to be updated immediately.

**Comment by alexchandel** [comment]
Title should be changed to "high latency."

FWIW I do experience sluggishness as well, and barely get 1 frame per second despite decent hardware, "Fastest" graphics, and only 1680x1050 game resolution.

**Comment by JapaMala** [comment]
How much of the fortress are you viewing at once? If you increase the viewing area, the performance drops quickly.

**Comment by alexchandel** [comment]
Maybe a 24x24 area? It also happens in FPS mode, though after several minutes in a particular area the framerate increases to ~10/s, around the time AV's CPU usage drops from 200% to 30%.

Would be nice to have a "Diagnostic HUD" that displayed the framerate, viewer coordinates, rendered area, & other diagnostic information, similar to Minecraft's `F3` menu.

**Comment by alexchandel** [comment]
My parameters are 5 up, 4 side, and 5 down. In FPS mode, if I descend a ladder more than about 6 layers below the layer I began on, Armok Vision crawls to a halt, at <1 frame/second. However, as soon as I ascend back, the framerate rises to a usable value.

**Comment by JapaMala** [comment]
Huh. Is it that specific area, or just any time you're going down?

I know generating new meshes does slow things down, but it shouldn't be that much.

**Comment by alexchandel** [comment]
Seems to be any time I descend far enough from where I spawn, but I only have one staircase so it's hard to compare.

Could be a platform-specific issue. Do you have a machine to test the mac build?

**Comment by JapaMala** [comment]
What happens if you don't do it in first person mode?

**Comment by alexchandel** [comment]
It appears to be worse in FPS mode. It's hard to tell, but the lag in god mode when I descend seems to cause an initial hang, where it skips over a few layers, then responds more (still slow, but above 1 frame per second).

**Comment by alexchandel** [comment]
I'm curious if the next release of Armok Vision, with the timers that'll presumably ameliorate #60, will also help my <1 FPS issue. If so, that could mean the FPS issue is caused by the network code (or mesh update code) running very poorly on Macs.

**Comment by alexchandel** [comment]
As of 85b649c93aaeb27ad42d46eb86e4b661f3eeb883 (i.e. just after 0.19.1), even with the reduced update timers, with AV now averaging ~240 KB/s in network reads when nothing is happening, the sluggishness remains, including about a 1-second delay in input response in FS mode.
